### PR TITLE
Update cuda to 11.7.1 for gcc12

### PR DESCRIPTION
**Pull request by aandvalenzuela**
See https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html

### CODE REVIEW

**Comment by aandvalenzuela** [comment]
enable gpu

**Comment by aandvalenzuela** [comment]
test parameters:
- full_cmssw = true

**Comment by aandvalenzuela** [comment]
please test

**Comment by aandvalenzuela** [comment]
enable gpu

**Comment by aandvalenzuela** [comment]
test parameters:
- full_cmssw = true

**Comment by aandvalenzuela** [comment]
please test

**Comment by aandvalenzuela** [comment]
Although updating `cudnn` to `8.4.1.50` for `cuda 11.7` (https://developer.nvidia.com/rdp/cudnn-archive), the correct url as `11.6` as default `cuda` version (https://developer.nvidia.com/compute/cudnn/secure/8.4.1/local_installers/11.6/cudnn-linux-x86_64-8.4.1.50_cuda11.6-archive.tar.xz).

**Comment by aandvalenzuela** [comment]
enable gpu

**Comment by aandvalenzuela** [comment]
test parameters:
- full_cmssw = true

**Comment by aandvalenzuela** [comment]
please test

File: cudnn.spec [hunk 1]
```diff
@@ -12,9 +14,9 @@
 %endif
 
 # cuDNN archive base name and unpacked name
-%define archive cudnn-linux-%{nvarch}-%{realversion}_cuda%{cudaver}-archive
+%define archive cudnn-linux-%{nvarch}-%{realversion}_cuda%{default_cudaver}-archive
 
-Source: https://developer.download.nvidia.com/compute/redist/cudnn/v%{cudnnver_maj}/local_installers/%{cudaver}/%{archive}.tar.xz
+Source: https://developer.download.nvidia.com/compute/redist/cudnn/v%{cudnnver_maj}/local_installers/%{default_cudaver}/%{archive}.tar.xz
```
Comment (smuzaffar) [inline cudnn.spec:19]: ah ok, so `cudnn` is not yet available for `cuda 11.7`

**Comment by fwyzard** [comment]
> See https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html

CUDA 11.7 Update 1 adds official support for RHEL 9, but it does not support GCC 12, only GCC 11.

See https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#system-requirements .

**Comment by fwyzard** [comment]
As a side note, could you please tag @cms-sw/heterogeneous-l2 when there are changes to externals related to GPUs (cuda, rocm, alpaka, and indirectly eigen) ?

**Comment by smuzaffar** [comment]
> As a side note, could you please tag @cms-sw/heterogeneous-l2 when there are changes to externals related to GPUs (cuda, rocm, alpaka, and indirectly eigen) ?

sure, I will see if bot can do this automatically

**Comment by smuzaffar** [comment]
> > See https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html
> 
> CUDA 11.7 Update 1 adds official support for RHEL 9, but it does not support GCC 12, only GCC 11.
> 
> See https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#system-requirements .

PR was not intended to go in, it was just to see if `cuda 11.7`  fixes/supports gcc 12

**Comment by smuzaffar** [comment]
closing this as cuda/gcc12 suppot is still missing. Also cudnn does not yes support cuda 11.7

**Pull request closed without merging**